Update swift coverage conversion script for Xcode 13 and 14 (#151)
SonarSource/sonar-scanning-examples: swift-coverage/README.md

# SonarQube-Scanner for Swift Code Coverage

> This example demonstrates how to import Xcode Coverage data (aka ProfData) to SonarQube for a Swift project. It supports Xcode 13 and above.

Author: Done

## Prerequisites

* [SonarQube](http://www.sonarqube.org/downloads/) 9.8 or Latest

> * [SonarQube Scanner](http://docs.sonarqube.org/display/SCAN/Analyzing+with+SonarQube+Scanner) 4.8+

Author: Reverted

# SonarQube-Scanner for Swift Code Coverage

> This example demonstrates how to import Xcode Coverage data to SonarQube for a Swift project. It supports Xcode 13 and above.

Reviewer: `13` -> `13.3`
And I would move this to the prerequisites section

xcodebuild -project swift-coverage-example.xcodeproj/ -scheme swift-coverage-example -derivedDataPath Build/ -enableCodeCoverage YES clean build test CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO
```

1.b Using xccov (recommended)

The `xccov` command line tool is the recommended option to view Xcode coverage
data and is more straightforward to use than the older `llvm-cov` tool. With
the script `xccov-to-sonarqube-generic.sh`, you can convert Xcode test results
stored in `*.xcresult` folders to the [SonarQube generic test coverage format](https://docs.sonarqube.org/latest/analyzing-source-code/test-coverage/generic-test-data/).

> First, locate the Xcode test result folder and convert the coverage data to the
> SonarQube format using the script as in the following example:

Reviewer: ```suggestion
First, locate the Xcode test result folder (`*.xcresult`). Then use it as a parameter to the script converting the coverage data to the SonarQube format as in the following example:
```

```shell

> bash xccov-to-sonarqube-generic.sh Build/Logs/Test/Run-swift-coverage-example-2023.01.27_16-07-44-+0100.xcresult >Coverage.xml

Reviewer: let's add `/` after`.xcresult` to be explicit that it's kind of dir :)

```

Then, use the parameter `sonar.coverageReportPaths` to reference the generated report:

```shell
sonar-scanner -Dsonar.coverageReportPaths=Coverage.xml
```

This parameter accepts a comma-separated list of files, which means you can also provide multiple coverage reports from multiple test results.

1.b Using llvm-cov 

You can also provide code coverage data using the `llvm-cov` format. The
process of generating an llvm-cov report requires several steps to get the
coverage for the application executable and the dynamic library binaries.

> In the case of the project example, first, locate the `Coverage.profdata` file
> under the `ProfileData` folder. Then, generate an `llvm-cov` report as in the
> following example:

Reviewer: ```suggestion
In the case of the project example, first, locate the `Coverage.profdata` file
under the `ProfileData` folder. Then, generate an `llvm-cov` report as in the
following example (the located `Coverage.profdata` file should be the value of `-instr-profile` parameter):
```
